homeassistant/components/media_player/openhome.py

```diff
@@ -0,0 +1,223 @@
+"""
+Support for Openhome Devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.openhome/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK, SUPPORT_TURN_ON,
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_SET, SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_STEP, SUPPORT_STOP, SUPPORT_PLAY, SUPPORT_SELECT_SOURCE,
+    MediaPlayerDevice)
+from homeassistant.const import (
+    STATE_IDLE, STATE_PAUSED, STATE_PLAYING, STATE_OFF)
+
+REQUIREMENTS = ['openhomedevice==0.2']
+
+SUPPORT_OPENHOME = SUPPORT_SELECT_SOURCE | \
+    SUPPORT_VOLUME_STEP | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | \
+    SUPPORT_TURN_OFF | SUPPORT_TURN_ON
+
+_LOGGER = logging.getLogger(__name__)
+
+# List of devices that have been registered
+DEVICES = []
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Openhome Platform."""
+    from openhomedevice.Device import Device
+
+    if discovery_info:
+        _LOGGER.info('Openhome device found, (%s)', discovery_info[0])
+        device = Device(discovery_info[1])
+
+        # if device already exists by config
+        if device.Uuid() in [x.unique_id for x in DEVICES]:
+            return True
+
+        device = HassOpenhomeDevice(hass, device)
+
+        add_devices([device], True)
+        DEVICES.append(device)
+
+        return True
+
+    return True
+
+
+class HassOpenhomeDevice(MediaPlayerDevice):
+    """Representation of an Openhome device."""
+
+    def __init__(self, hass, device):
+        """Initialise the Openhome device."""
+        self.hass = hass
+        self._device = device
+        self._track_information = {}
+        self._in_standby = None
+        self._transport_state = None
+        self._track_information = None
+        self._volume_level = None
+        self._volume_muted = None
+        self._supported_features = SUPPORT_OPENHOME
+        self._source_names = list()
+        self._name = device.Uuid()
+        self._state = STATE_PLAYING
+
+        self.update()
```

Comment:
This `self.update` isn't necessary, since you call `add_devices` with `True`, which will call the update itself: https://github.com/home-assistant/home-assistant/pull/6153/files#diff-80cd99321d808e4726a56be704e3085eR44